### PR TITLE
Bug fix: Replace `hasValue` with `isAnswered` methods throughout applicant question classes

### LINKED ISSUE #634: `hasValue` in the `ApplicantQuestion` subclasses don't seem to WAI
We decided on a convention of storing `null` for "never answered" pieces of data, and `""` (the empty string) for "answered with a blank value".

However, it seems like the underlying spring form data binder binds empty inputs as `null` (at least that's what I guess is happening).

I ran into this issue while implementing min and max length validations for text questions. When there is no value entered in the applicant form, our code evaluates `hasValue()` to `false`, so we end up not running our `getQuestionErrors` validations.

Needs more investigation to conclude the root cause and obviously to fix the issue as well.

### PR DESCRIPTION
**Pull request by natsid**
### Description
Fixes a bug. In some cases, we were depending on `hasValue` to determine whether or not a question had an attempted answer, which would then help determine whether or not the question had any errors. However, `hasValue` would return false, even if there was an attempted answer, but it was empty. Instead, we use an `isAnswered` method that checks whether the JSON path exists in the applicant data. Now, even if the answer is empty, `isAnswered` will return true if the path exists.

### Note
This still doesn't work perfectly. See https://github.com/seattle-uat/civiform/issues/783 for why. Linked TODOs where relevant.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #634

### CODE REVIEW

File: universal-application-tool-0.0.1/app/services/applicant/question/MultiSelectQuestion.java [hunk 1]
```diff
@@ -61,6 +61,10 @@ public boolean hasValue() {
     return getSelectedOptionsValue().isPresent();
   }
 
+  public boolean isAnswered() {
```
Comment (kendrickt) [inline universal-application-tool-0.0.1/app/services/applicant/question/MultiSelectQuestion.java:64]: (blocking) Should this be added to the interface? or should there be an additional interface if `PresentsErrors` isn't the right scope?

It seems like if every `SomethingQuestion` has a public `isAnswered` method it should be available generically.

Comment (natsid) [reply universal-application-tool-0.0.1/app/services/applicant/question/MultiSelectQuestion.java:64]: Good point.... I think that sounds good. The name that comes to mind is `Answerable`. LMK if you think of something better.

**Review by hmhsu — APPROVED** [review]

File: universal-application-tool-0.0.1/app/services/applicant/question/MultiSelectQuestion.java (hunk 1 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/services/applicant/question/MultiSelectQuestion.java:64]: Hmm, I guess one awkward aspect of it is that multi-field question types, like `AddressQuestion` have multiple `Answered` methods, like `streetAnswered`, `cityAnswered`. We could have an overall `isAnswered` method that takes the union of those though.

I'm trying to think of what value it adds to have an interface and the overall `isAnswered` method. The only place we really use it is to determine whether or not to check for errors. With that in mind, maybe it does make the most sense to have `isAnswered` be part of the `PresentsErrors` interface.

Comment (natsid) [reply universal-application-tool-0.0.1/app/services/applicant/question/MultiSelectQuestion.java:65]: Updated. Let me know what you think of the change, @kendrickt .

**Comment by natsid** [comment]
@kendrickt and I chatted offline yesterday.

As a result, I created https://github.com/seattle-uat/civiform/issues/783 and linked this issue in TODOs throughout these applicant question classes to use the "hydrated" paths instead of the generic question definition paths.

**Review by kendrickt — APPROVED** [review]
lgtm

**Pull request merged**